src/persist-client/src/read.rs

```diff
@@ -173,33 +189,107 @@ where
         }
     }
 
-    /// Attempt to pull out the next values of this subscription. For more
-    /// details, see [`Listen::next`].
-    ///
-    /// TODO: delete this method when refactoring
-    /// `storage::source::persist_source::persist_source_sharded`.
-    #[instrument(level = "debug", skip_all, fields(shard = %self.listen.handle.machine.shard_id()))]
-    pub async fn next_listen_events(&mut self) -> Vec<ListenEvent<K, V, T, D>> {
-        trace!(
-            "Subscribe::next as_of={:?}, frontier={:?}",
-            self.listen.as_of,
-            self.listen.frontier
-        );
-        // This is odd, but we move our handle into a `Listen`.
-        self.listen.handle.maybe_heartbeat_reader().await;
+    /// Passes through request to drop SeqNo lease
+    pub fn drop_seqno_lease(&mut self, seqno: SeqNo) {
+        self.listen.handle.drop_seqno_lease(seqno);
+    }
+}
 
-        match self.snapshot_batches.pop_front() {
-            Some(batch) => {
-                let updates = self
-                    .listen
-                    .handle
-                    .fetch_batch(batch)
-                    .await
-                    .expect("must accept self-generated batch");
-                vec![ListenEvent::Updates(updates)]
-            }
-            None => self.listen.next().await,
+/// Capable of fetch batches, and appropriately handling the metadata on
+/// [`ReaderEnrichedHollowBatch`]es.
+#[derive(Debug)]
+pub struct BatchFetcher<K, V, T, D>
```

Comment:
nit: move this (along with fetch_batch_part) to a new `mod fetch` please

Reply:
agreed.

Also: I like this refactoring! :raised_hands:

Reply:
Done in commit w/ message `persist: add fetch submodule to read`